### ISSUE TITLE
Optimize uuid codecs more

### ISSUE THREAD

**Issue by fabiolimace** [f4b6a3/uuid-creator#47]
1. Memory usage optimization: only the base-16 is required by the lib. The other base-n instances should be loaded if the user needs them.

2. Processing optimization: add base-n codecs for base-16, base-32 and base-36 that encode do UPPER CASE without calling `String.toUpperCase()`.

3. Add more test cases.

**Comment by fabiolimace** [comment]
Removed public static final instances of `BaseN`. It is not necessary to load all base-n instances at once. Only base-16 is required by lib. The other base-n instances should only be instantiated if the user needs a `BaseNCodec` other than `Base16Codec`. This is a memory usage optimization.

Commit: 8942d59388d528f70bf95da71890718cb4494ed9

**Comment by fabiolimace** [comment]
It was necessary to use `String.toUpperCase()` to get a UPPER CASE base-16 string.

```java
String hexa = Base16Codec.INSTANCE.encode(uuid).toUpperCase();
```

Now the user can choose a codec that encodes to UPPER CASE.

```java
String hexa = Base16UCodec.INSTANCE.encode(uuid);
```

This is a cpu usage optimization in the case of upper case base-n.

Commit: e64bfc38c9fa83dfbb3f52e77f25f0d961fd6762

**Comment by fabiolimace** [comment]
Released version 4.1.1.
